### ISSUE TITLE
App doesn't load and Crashes

### ISSUE THREAD

**Issue by girlabroad** [nick92/coffee#31]
Just installed on Linux Mint 18.1 64b
running Cinnamon 3.2.7
Kernal 4.4.0-150-generic

nothing loads and the loading wheel just spins.  When i try to manually add a location the app just closes completely.

**Comment by nick92** [comment]
if you navigate into the settings do you have anything selected?

**Comment by girlabroad** [comment]
Yes, of course.  I tried turning all but one source off as well, in case that was slowing it.
Restarted machine.
It just spins and spins and just closes when you try to select a location

**Comment by nick92** [comment]
are you on the latest version? 

and could you run it from the terminal by running com.github.nick92.coffee -d and pasting the output into here

**Comment by girlabroad** [comment]
version was downloaded directly from the version on the website.  
pasted that command in and it says :
unknown option -d 

i ran sudo apt-get upgrade and it wasn't listed.

I'm no linux expert so handle with care :)

**Comment by nick92** [comment]
I've updated the version of coffee for 16.04 now so if you run 

sudo apt-get update && sudo apt-get upgrade and let me know if you have any further issues with it

**Comment by girlabroad** [comment]
Hi
The news and stories now update but when i try to put in a weather location it just closes.  I'm guessing you don't have my location in the list.  Under the automatic weather setting it's giving me weather for a city over 200 km's away.

**Comment by nick92** [comment]
could be due to your location, what is it? I'll give a try

**Comment by girlabroad** [comment]
Kappeln in Northern Germany
nearest large town would be Eckernförder and the nearest city is Flensburg

**Comment by nick92** [comment]
That works for me, so I go into settings -> weather and turn off auto location and then put Kappeln into the location textbox and then hit enter, I refresh the news view and it loads up the weather for that location 

could you provide steps of what you are doing any maybe a screenshot just before it crashes?

**Comment by girlabroad** [comment]
what's exactly what i did.
went to settings
turned off auto location
entered kappeln
bam - gone

just tried it one more time for luck and same.

**Comment by nick92** [comment]
could you run com.github.nick92.coffee -d and paste the output into here?

**Comment by girlabroad** [comment]
[DEBUG 15:11:04.727518] [Gtk] Connecting to session manager
[DEBUG 15:11:06.234161] [CoffeeBar:183] Added all weather at : 1.5030220000000001 s
[DEBUG 15:11:06.235045] [CoffeeBar:124] Got Weather at: 1.5039849999999999 s
[DEBUG 15:11:06.843725] [NewsItem:128] HTTP Error: Bad Gateway
[CRITICAL 15:11:06.849000] [Gtk] gtk_container_add: assertion 'GTK_IS_WIDGET (widget)' failed
[DEBUG 15:11:10.051337] [CoffeeBar:160] Added all posts at : 5.3202689999999997 s
[DEBUG 15:11:10.051457] [CoffeeBar:114] Got News at: 5.3204190000000002 s
[CRITICAL 15:12:19.909673] [Gtk] gtk_combo_box_get_active_id: assertion 'GTK_IS_COMBO_BOX (combo_box)' failed
[CRITICAL 15:12:19.919307] settings_news_news_source_get_get_sources: assertion 'country != NULL' failed
[DEBUG 15:12:20.762276] [NewsContainer:146] Getting icon from Besticon:http://www.afr.com
[DEBUG 15:12:20.762481] [NewsSourcesGet:79] get_besticon_url(http://www.afr.com)
[DEBUG 15:12:20.767084] [NewsContainer:146] Getting icon from Besticon:http://www.bloomberg.com
[DEBUG 15:12:20.767208] [NewsSourcesGet:79] get_besticon_url(http://www.bloomberg.com)
[DEBUG 15:12:20.770364] [NewsContainer:146] Getting icon from Besticon:http://www.businessinsider.com
[DEBUG 15:12:20.772445] [NewsContainer:146] Getting icon from Besticon:http://uk.businessinsider.com
[DEBUG 15:12:20.774365] [NewsContainer:146] Getting icon from Besticon:http://www.cnbc.com
[DEBUG 15:12:20.776285] [NewsContainer:146] Getting icon from Besticon:http://business.financialpost.com
[DEBUG 15:12:20.778134] [NewsContainer:146] Getting icon from Besticon:http://fortune.com
[DEBUG 15:12:20.782266] [NewsContainer:146] Getting icon from Besticon:http://www.economist.com
[DEBUG 15:12:20.787091] [NewsContainer:146] Getting icon from Besticon:http://www.wsj.com
[WARN 15:12:20.790924] [NewsGrid:125] complete
[CRITICAL 15:12:21.039127] [Json] json_array_get_object_element: assertion 'index_ < array->elements->len' failed
[CRITICAL 15:12:21.049049] [Json] json_object_get_string_member: assertion 'object != NULL' failed
Segmentation fault
